src/peergos/shared/user/fs/FileWrapper.java

```diff
@@ -1450,15 +1457,16 @@ public SigningPrivateKeyAndPublicHash signingPair() {
 
         return context.network.synchronizer.applyComplexUpdate(target.owner(), target.signingPair(),
                 (version, committer) -> version.withWriter(owner(), writer(), network)
-                        .thenCompose(both -> copyTo(target, network, crypto, both, committer)))
+                        .thenCompose(both -> copyTo(target, network, crypto, both, committer, this.props.thumbnail)))
                 .thenApply(newAccess -> true);
     }
 
     public CompletableFuture<Snapshot> copyTo(FileWrapper target,
                                               NetworkAccess network,
                                               Crypto crypto,
                                               Snapshot version,
-                                              Committer committer) {
+                                              Committer committer,
+                                              Optional<Thumbnail> existingThumbnail) {
```

Comment:
The convention I have at the moment is to keep "meaningful" parameters before things like network, crypto, version and committer. Could we put exisitngThumnail as second param?